packages/online-editor/src/accelerators/AcceleratorsContext.tsx

```diff
@@ -0,0 +1,414 @@
+            </>
+          }
+        />
+      ),
+      [i18n]
+    )
+  );
+
+  const applyAcceleratorSuccessAlert = useGlobalAlert(
+    useCallback(
+      (_, staticArgs: { acceleratorName: string }) => (
+        <Alert
+          className={"kogito--alert"}
+          variant="success"
+          title={i18n.accelerators.successAlert(staticArgs.acceleratorName)}
+        />
+      ),
+      [i18n]
+    ),
+    { durationInSeconds: 4 }
+  );
+
+  const applyAcceleratorFailAlert = useGlobalAlert(
+    useCallback(
+      (_, staticArgs: { acceleratorName: string }) => (
+        <Alert
+          className={"kogito--alert"}
+          variant="danger"
+          title={i18n.accelerators.failAlert(staticArgs.acceleratorName)}
+        />
+      ),
+      [i18n]
+    ),
+    { durationInSeconds: 4 }
+  );
+
+  const attemptToDeleteTemporaryBranches = useCallback(async () => {
+    const workspaceId = workspace.descriptor.workspaceId;
+    try {
+      await workspaces.deleteBranch({ workspaceId, ref: BACKUP_BRANCH_NAME });
+    } catch (e) {
+      // Do nothing
+    }
+    try {
+      await workspaces.deleteBranch({ workspaceId, ref: MOVED_FILES_BRANCH_NAME });
+    } catch (e) {
+      // Do nothing
+    }
+  }, [workspace.descriptor.workspaceId, workspaces]);
+
+  const applyAcceleratorToWorkspace = useCallback(
+    async (accelerator: AcceleratorConfig, currentFile: WorkspaceFile) => {
+      apllyingAcceleratorAlert.show({ acceleratorName: accelerator.name });
+
+      const workspaceId = workspace.descriptor.workspaceId;
+
+      let movedFiles: WorkspaceFile[] = [];
+
+      // Attempt to delete branches
+      // This is a workaround, as isomorphic-git doesn't necessarilly deletes the branch, even though it resolves
+      attemptToDeleteTemporaryBranches();
+
+      let configFile: WorkspaceFile | undefined = undefined;
+
+      try {
+        const destinationPathValidation = {
+          dmnDestinationFolder: validateAcceleratorDestinationFolderPaths(accelerator.dmnDestinationFolder),
+          bpmnDestinationFolder: validateAcceleratorDestinationFolderPaths(accelerator.bpmnDestinationFolder),
+          otherFilesDestinationFolder: validateAcceleratorDestinationFolderPaths(
+            accelerator.otherFilesDestinationFolder
+          ),
+        };
+
+        if (
+          destinationPathValidation.dmnDestinationFolder ||
+          destinationPathValidation.bpmnDestinationFolder ||
+          destinationPathValidation.otherFilesDestinationFolder
+        ) {
+          throw destinationPathValidation;
+        }
+
+        const workspaceFiles = await workspaces.getFiles({ workspaceId });
+
+        // Create new temporary branch with current files, but stay on main
+        await workspaces.branch({ workspaceId, name: BACKUP_BRANCH_NAME, checkout: false });
+
+        // Commit moved files to moved files branch
+        await workspaces.commit({
+          workspaceId,
+          commitMessage: "Backup files",
```

Comment:
```suggestion
          commitMessage: "KIE Sandbox: backup files before applying Accelerator ${acceleratorName}",
```

For illustrative purposes only, this code is not correct ^

Reply:
Sounds good, but this commit will never be seen/pushed anywhere

Reply:
Only when there's a bug and we need to understand what happened :P